pkg/sshd/lock.go

```diff
@@ -0,0 +1,68 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string) error
```

Comment:
@aledbf can this and `Unlock` below also accept a `timeout time.Duration`, so that thie future implementations can wait to acquire the lock